org.openscenegraph.android/jni/JNIosg.cpp

```diff
@@ -139,6 +178,931 @@ JNIEXPORT jint JNICALL Java_org_openscenegraph_osg_core_Group_nativeGetNumChildr
+{
+	bool result = false;
+	osg::Geode* g = reinterpret_cast<osg::Geode*>(cptr);
+	osg::Drawable* d = reinterpret_cast<osg::Drawable*>(cptr_drawable);
+	if(g!=NULL)
+	{
+		result = g->removeDrawable(d);
+	}
+	return result?JNI_TRUE:JNI_FALSE;
+}
+
+JNIEXPORT jboolean JNICALL Java_org_openscenegraph_osg_core_Geode_nativeRemoveDrawables(JNIEnv* env, jclass, jlong cptr, jint i, jint m)
+{
+	bool result = false;
+	osg::Geode* g = reinterpret_cast<osg::Geode*>(cptr);
+	if(g!=NULL)
+	{
+		result = g->removeDrawables((unsigned int)i, (unsigned int)m);
+	}
+	return result?JNI_TRUE:JNI_FALSE;
+}
+
+JNIEXPORT jboolean JNICALL Java_org_openscenegraph_osg_core_Geode_nativeReplaceDrawable(JNIEnv* env, jclass, jlong cptr, jlong cptr_drawable_old, jlong cptr_drawable_new)
+{
+	bool result = false;
+	osg::Geode* g = reinterpret_cast<osg::Geode*>(cptr);
+	osg::Drawable* d_old = reinterpret_cast<osg::Drawable*>(cptr_drawable_old);
+	osg::Drawable* d_new = reinterpret_cast<osg::Drawable*>(cptr_drawable_new);
+	if(g!=NULL)
+	{
+		result = g->replaceDrawable(d_old,d_new);
+	}
+	return result?JNI_TRUE:JNI_FALSE;
+}
+
+JNIEXPORT jboolean JNICALL Java_org_openscenegraph_osg_core_Geode_nativeSetDrawable(JNIEnv* env, jclass, jlong cptr, jint i, jlong cptr_drawable)
+{
+	bool result = false;
+	osg::Geode* g = reinterpret_cast<osg::Geode*>(cptr);
+	osg::Drawable* d = reinterpret_cast<osg::Drawable*>(cptr_drawable);
+	if(g!=NULL)
+	{
+		result = g->setDrawable((unsigned int)i, d);
+	}
+	return result?JNI_TRUE:JNI_FALSE;
+}
+
+JNIEXPORT jint JNICALL Java_org_openscenegraph_osg_core_Geode_nativeGetNumDrawables(JNIEnv* env, jclass, jlong cptr)
+{
+	int result = 0;
+	osg::Geode* g = reinterpret_cast<osg::Geode*>(cptr);
+	if(g!=NULL)
+	{
+		result = g->getNumDrawables();
+	}
+	return reinterpret_cast<jint>(result);
+}
+
+JNIEXPORT jlong JNICALL Java_org_openscenegraph_osg_core_Geode_nativeGetDrawable(JNIEnv* env, jclass, jlong cptr, jint i)
+{
+	jlong result = 0;
+	osg::Geode* g = reinterpret_cast<osg::Geode*>(cptr);
+	if(g!=NULL)
+	{
+		osg::Drawable* d = g->getDrawable((int)i);
+		result = reinterpret_cast<jlong>(d);
+	}
+	return result;
+}
+
+/*
+ * osg::Geometry
+ */
+JNIEXPORT void JNICALL Java_org_openscenegraph_osg_core_Geometry_nativeDispose(JNIEnv *, jclass, jlong cptr)
+{
+	osg::Geometry *g = reinterpret_cast<osg::Geometry *>(cptr);
+	if(g!=NULL)
+		g->unref();
+}
+
+JNIEXPORT jlong JNICALL Java_org_openscenegraph_osg_core_Geometry_nativeCreateGeometry(JNIEnv *, jclass)
+{
+    osg::Geometry *g = new osg::Geometry();
+    g->ref();
+    return reinterpret_cast<jlong>(g);
+}
+
+JNIEXPORT void JNICALL Java_org_openscenegraph_osg_core_Geometry_nativeSetVertexArray(JNIEnv* env, jclass, jlong cptr, jobjectArray inputBuffer)
+{
+	osg::Geometry *g = reinterpret_cast<osg::Geometry*>(cptr);
+
+	int array_n = env->GetArrayLength(inputBuffer);
+	//printf("Array size: %i\n", array_n);
+	osg::Vec3Array* _array = new osg::Vec3Array();
```

Comment:
We prefer local variables name without underscore, change "_array" to "array". We only use the underscore in class attributes.